### ISSUE TITLE
Node 4.x support

### ISSUE THREAD

**Issue by springmeyer** [mapbox/node-pre-gyp#173]
Node 4.x mostly works, but will keep this open until all tests are passing and we're able to expose in the travis testing matrix.

Currently 4.x is broken in testing by `gyp ERR! stack Error: Invalid version number: -Dfoo=bar` which is due to https://github.com/nodejs/node-gyp/issues/720.

**Comment by ChALkeR** [comment]
Adding to the list: https://github.com/nodejs/node/issues/2798.

**Comment by springmeyer** [comment]
https://github.com/nodejs/node-gyp/issues/720 is fixed so the node 4.1.x release should avoid that problem. Testing 4.1.x I'm now seeing:

```
6) complex builds app2 app2 via remote passes tests --custom_include_path=../include --debug:
     Uncaught AssertionError: 'npm WARN package.json node-pre-gyp-test-app2@0.1.0-release1.release2+build1.build2 No license field.\n' == ''
```

refs https://github.com/npm/npm/issues/5253. Trying to workaround like https://github.com/mapbox/node-pre-gyp/commit/bf0e093c6b333cec6db2f1b1c59415e2572d662f

**Comment by springmeyer** [comment]
bf0e093, closing.
